metashare/api/jobs.py

```diff
@@ -71,6 +77,44 @@ def _create_branches_on_github(*, user, repo_id, project, task):
     return task_branch_name
 
 
+def alert_user_about_expiring_org(org, user):
+    from .models import User, ScratchOrg
+
+    def _do_alert():
+        # if scratch org is there
+        try:
+            org.refresh_from_db()
+            user.refresh_from_db()
+        except (ScratchOrg.DoesNotExist, User.DoesNotExist):
+            return
+
+        # and has unsaved changes
+        if org.unsaved_changes:
```

Comment:
Remind me, when does this get updated?

Reply:
In `ScratchOrg.finalize_get_unsaved_changes` if there's an error (reverts to `{}`), in `jobs.get_unsaved_changes`, and in `jobs.commit_changes_from_org`. In the case of the two jobs, it gets set to the value of `compare_revisions`.

`get_unsaved_changes` is kicked off in the background when we hit the list or retrieve endpoints for scratch orgs, assuming a list of six conditions are met for that scratch org.

`commit_changes_from_org` is kicked off when we hit the scratch org commit endpoint.